src/Middleware/src/Headstart.Common/Services/CMS/IAssetClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Headstart.Common.Extensions;
+using Headstart.Common.Services.CMS.Models;
+using Headstart.Models;
+using ordercloud.integrations.library;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Headstart.Common.Services.CMS
+{
+    public interface IAssetClient
+    {
+        Task<ImageAsset> CreateImage(AssetUpload asset);
+        Task DeleteAsset(string id);
+        Task<DocumentAsset> CreateDocument(AssetUpload asset);
+        Task DeleteAssetByUrl(string assetUrl);
+    }
+
+    public class AssetClient : IAssetClient
+    {
+        private readonly IOrderCloudIntegrationsBlobService _blob;
+        private readonly AppSettings _settings;
+
+        public AssetClient(IOrderCloudIntegrationsBlobService blob, AppSettings settings)
+        {
+            _blob = blob;
+            _settings = settings;
+        }
+
+        public async Task<ImageAsset> CreateImage(AssetUpload asset)
+        {
+            var container = _blob.Container.Name;
+            var assetGuid = Guid.NewGuid().ToString();
+
+            using(var image = Image.FromStream(asset.File.OpenReadStream()))
+            {
+                var small = image.ResizeSmallerDimensionToTarget(100);
+                var medium = image.ResizeSmallerDimensionToTarget(300);
+                try
+                {
+                    await Task.WhenAll(new[] {
+                        _blob.Save(assetGuid, medium.ToBytes(ImageFormat.Png), "image/png"),
+                        _blob.Save($"{assetGuid}-s", small.ToBytes(ImageFormat.Png), "image/png")
+                    });
+                } finally
+                {
+                    small.Dispose();
+                    medium.Dispose();
+                }
+            }
```

Comment:
You don't need to manually dispose. Since you're wrapping this in a `using` statement, it will get disposed for you